### PR TITLE
Add RabbitMQ Alerts 🚨

### PR DESCRIPTION
**Pull request by ansd**
Add Prometheus rules for RabbitMQ and cluster operator.
Add Alertmanager config for Slack notifications.
Add RabbitMQ-Alerts Grafana dashboard.

Blog post on https://blog.rabbitmq.com/ will follow soon.

In future, we might want to move all dashboards from https://github.com/rabbitmq/rabbitmq-server/tree/master/deps/rabbitmq_prometheus/docker/grafana/dashboards into this repository and create GitHub actions to publish dashboards on https://grafana.com/orgs/rabbitmq.

### CODE REVIEW

**Review by MirahImage — APPROVED** [review]

File: observability/prometheus/rules/rabbitmq/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# RabbitMQ Prometheus Rules
+
+## Adding new Prometheus Rules
+
+To allow filtering in the [RabbitMQ-Alerts Grafana dashboard](../../../grafana/dashboards/rabbitmq-alerts.yml) and allowing [configuration of Alertmanager](../../alertmanager) all RabbitMQ alerts must output at least the labels:
+* `rulesgroup: rabbitmq`
+* `namespace`
+* `rabbitmq_cluster`
+* `severity`
```
Comment (coro) [inline observability/prometheus/rules/rabbitmq/README.md:9]: None of these rules have all of these; am I missing something?

File: observability/prometheus/rules/rabbitmq/insufficient-established-erlang-distribution-links.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      expr: |
+        count by (namespace, rabbitmq_cluster) (erlang_vm_dist_node_state * on(instance) group_left(rabbitmq_cluster) rabbitmq_identity_info == 3)
+        <
+         count by (namespace, rabbitmq_cluster) (rabbitmq_build_info * on(instance) group_left(rabbitmq_cluster) rabbitmq_identity_info)
+         *
+        (count by (namespace, rabbitmq_cluster) (rabbitmq_build_info * on(instance) group_left(rabbitmq_cluster) rabbitmq_identity_info) -1 )
+      for: 10m
+      annotations:
+        description: |
+          The are only `{{ $value }}` established Erlang distribution links
```
Comment (coro) [inline observability/prometheus/rules/rabbitmq/insufficient-established-erlang-distribution-links.yml:24]: Typo: `There`

**Review by coro — APPROVED** [review]
Looks amazing!
image

I love that you include remediation/next steps n each of the alerts, I think that's very useful for someone hitting these alerts and not knowing how to fix them.

I'm curious what you would/did use to test these alerts, and if it's worth sharing that tooling.

File: observability/prometheus/rules/rabbitmq/insufficient-established-erlang-distribution-links.yml (hunk 2 above)
Comment (ansd) [reply observability/prometheus/rules/rabbitmq/insufficient-established-erlang-distribution-links.yml]: Fixed, thanks.

File: observability/prometheus/rules/rabbitmq/README.md (hunk 1 above)
Comment (ansd) [reply observability/prometheus/rules/rabbitmq/README.md:9]: All rules output all these labels. In some rules, they are not explicitly added via the `labels` field but implicitly output by the PromQL expression. I added a sentence to the README which makes it clearer.

File: observability/prometheus/alertmanager/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Alertmanager
+
+The file [slack.yml](./slack.yml) contains [Alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/) configuration that sends Slack notifications for RabbitMQ and RabbitMQ Cluster Operator alerts.
+It groups alerts by namespace and RabbitMQ cluster.
+
+Replace
```
[the remark this answers is not in the document]
Comment (ansd) [reply observability/prometheus/alertmanager/README.md:6]: Yes :) As next step, when we play the Carvel story, we should use `ytt`.
For the start, I wanted to keep it as simple as possible. Therefore, the `quickstart.sh` uses `sed`.

File: observability/quickstart.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  printf "%bEnvironment variables SLACK_CHANNEL or SLACK_API_URL not set. " "$ORANGE"
+  printf "You will therefore not receive Slack notifications.%b\n" "$NO_COLOR"
+fi
+
+printf "%bInstalling RabbitMQ Grafana Dashboards...%b\n" "$GREEN" "$NO_COLOR"
+kubectl -n "$KUBE_PROMETHEUS_STACK_NAMESPACE" apply --filename "$DIR"/grafana/dashboards/
+
+printf "%bInstalling RabbitMQ Cluster Operator...%b\n" "$GREEN" "$NO_COLOR"
+kubectl apply --filename https://github.com/rabbitmq/cluster-operator/releases/latest/download/cluster-operator.yml
+#TODO ansd remove below line once cluster-operator >v1.6.0 is cut (which will include https://github.com/rabbitmq/cluster-operator/pull/664)
```
[the remark this answers is not in the document]
Comment (coro) [reply observability/quickstart.sh:62]: Ahhh, I read `>` as `>=`, my bad!

File: observability/quickstart.sh [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+kubectl -n rabbitmq-system set image deployment/rabbitmq-cluster-operator operator="rabbitmqoperator/cluster-operator-dev:1.6.0-build.10"
```
[the remark this answers is not in the document]
Comment (ansd) [reply observability/quickstart.sh:63]: Good idea. I added additional output for port-forwarding to Grafana, Prometheus, and Alertmanager.

File: observability/grafana/dashboards/rabbitmq-alerts.yml [hunk 6]
```diff
@@ -0,0 +1,669 @@
...
+                  {
+                    "id": "custom.width",
+                    "value": 319
+                  }
+                ]
+              },
+              {
+                "matcher": {
+                  "id": "byName",
+                  "options": "PersistentVolumeClaim"
```
[the remark this answers is not in the document]
Comment (ansd) [reply observability/grafana/dashboards/rabbitmq-alerts.yml:296]: Good point. The renaming from alert label (i.e. the metadata) to table column name takes place in https://github.com/rabbitmq/cluster-operator/blob/ae6e74e919d99dd1baa929080c77444094870f77/observability/grafana/dashboards/rabbitmq-alerts.yml#L380-L392. In your table output above only some alerts have a label `persistentvolumeclaim`. If an alert doesn't have such a label, the cell in the table will be empty.

I think if we observe that the table contains too many columns, we can either remap or omit certain columns or introduce an opaque metadata label as you suggest. As of now, the user can also filter the table by alert type (via the drop down at the top).

File: observability/prometheus/monitors/README.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
+# Prometheus Scrape Targets
+
+This directory contains Prometheus scrape targets.
+
+Check the `spec` of your installed Prometheus custom resource.
+In this example, let's assume your Prometheus spec contains the following fields:
```
[the remark this answers is not in the document]
Comment (coro) [reply observability/prometheus/monitors/README.md:6]: Happy to wait on the blog post.

**Pull request merged**